### PR TITLE
Refresh Lightgbm to v4.0.0

### PR DESCRIPTION
**Pull request by ArchanaShinde1**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: lightgbm-4.0.0/recipe/meta.yaml [hunk 1]
```diff
@@ -12,12 +12,11 @@ source:
   git_tag: v{{ version }}
   patches:
     # 02xx - GPU only patch specific to open-ce (maybe)
-    - 0201-Build-fix-for-cuda.patch                         #[build_type == 'cuda']
```
Comment (cdeepali) [inline lightgbm-4.0.0/recipe/meta.yaml:15]: Are these patches no longer required?

Comment (ArchanaShinde1) [reply lightgbm-4.0.0/recipe/meta.yaml:15]: Not needed during build time of lightgbm, Verified build on PPC and x86 platform too.

Comment (npanpaliya) [reply lightgbm-4.0.0/recipe/meta.yaml:15]: We've also removed patch for cuda compute capabilities. I hope we are using right compute inspite of that. Please verify it once, if not done already.

File: lightgbm-4.0.0/recipe/meta.yaml [hunk 2]
```diff
@@ -12,12 +12,12 @@ source:
...
-    - 0202-Fix-for-cuda-compute-capabilities-setting.patch  #[build_type == 'cuda']
-    - 0203-Fix-os-import-in-setup.py.patch
- 
+ #   - 0201-Build-fix-for-cuda.patch                         #[build_type == 'cuda']
+ #   - 0202-Fix-for-cuda-compute-capabilities-setting.patch  #[build_type == 'cuda']
+    - 0201-Cuda-12.2-support.patch                           #[build_type == 'cuda']
```
Comment (cdeepali) [inline lightgbm-4.0.0/recipe/meta.yaml]: Should the filter be set to  # [cudatoolkit == "12.2"] ?

**Review by cdeepali — APPROVED** [review]

**Pull request merged**